hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/BucketCache.java

```diff
@@ -934,6 +936,27 @@ void freeSpace(final String why) {
     }
     try {
       freeInProgress = true;
+
+      // Fetch the file names from the backing map.
```

Comment:
The main logic is to rely on the file names. Fetch the file names from the block cache keys.
Pass the list to a new DataTieringManager API to detect which of these files are cold-files and return the list of cold files.

Reply:
We can use the BlocksByFile instead of BackingMap if it seems more efficient.